operators/test/e2e/failure_test.go

```diff
@@ -62,8 +62,8 @@ func killNodeTest(t *testing.T, s stack.Builder, listOptions client.ListOptions,
 			{
 				Name: "Wait for pod to be deleted",
 				Test: helpers.Eventually(func() error {
-					_, err := k.GetPod(killedPod.Name)
-					if apierrors.IsNotFound(err) {
+					pod, err := k.GetPod(killedPod.Name)
+					if apierrors.IsNotFound(err) || killedPod.UID != pod.UID {
```

Comment:
We should first check if `err != nil && !apierrors.IsNotFound(err) ` before doing something with `pod`
I will fix that in an other commit